templates/ha.template.yaml

```diff
@@ -135,136 +185,217 @@ Resources:
       Architecture: x86_64
       VirtualizationType: hvm
       Owners: '309956199498'
-  HAMaster:
+  HAMain:
     Type: 'AWS::EC2::Instance'
     Properties:
-      ImageId: !Ref HAInstanceAMI
+      ImageId: !If 
+        - IsMarketplaceDeployment
+        - !FindInMap
+          - RegionMap
+          - !Ref 'AWS::Region'
+          - x8664
+        - !Ref HAInstanceAMI
       InstanceType: m4.xlarge
       KeyName: !Ref KeyPairName
       SecurityGroupIds:
         - !Ref HaaSecurityGroup
       SubnetId: !Select [ '0', !Ref SubnetIDs ]
-      UserData: !Base64
-        'Fn::Join':
-          - |+
+      UserData: !If
+        - IsMarketplaceDeployment
+        - !Base64
+            'Fn::Join':
+              - |+
+
+              - - '#!/bin/bash'
+                - cd /home/ec2-user/temp
+                - chmod +x get-haa.sh
+                - !Join
+                  - ' '
+                  - - sh get-haa.sh
+                    - '-u'
+                    - !Ref HAAUser
+                    - '-p'
+                    - !Ref HAAPassword
+                    - !If
+                      - TrialHAA
+                      - ''
+                      - !Join
+                        - ''
+                        - - '-l '
+                          - !Ref HAALicense
+        - !Base64
+            'Fn::Join':
+              - |+
 
-          - - '#!/bin/bash'
-            - yum update -y
-            - yum install -y wget
-            - 'wget https://raw.githubusercontent.com/UiPath/Infrastructure/main/Setup/get-haa.sh'
-            - chmod +x get-haa.sh
-            - !Join
-              - ' '
-              - - sh get-haa.sh
-                - '-u'
-                - !Ref HAAUser
-                - '-p'
-                - !Ref HAAPassword
-                - !If
-                  - TrialHAA
-                  - ''
-                  - !Join
-                    - ''
-                    - - '-l '
-                      - !Ref HAALicense
+              - - '#!/bin/bash'
+                - yum update -y
+                - yum install -y wget
+                - 'wget https://raw.githubusercontent.com/UiPath/Infrastructure/main/Setup/get-haa.sh'
```

Comment:
Why this script is being downloaded from outside git repo and not part of the Quick Start?

Reply:
Is this required now? I believe we have more external dependencies.